src/editor/controller/editor_controller.cpp

```diff
@@ -703,13 +711,6 @@ bool editor_controller::do_execute_command(const hotkey::ui_command& cmd, bool p
 
 			//Palette
 		case HOTKEY_EDITOR_PALETTE_GROUPS:
-		{
-			//TODO this code waits for the gui2 dialog to get ready
-//			std::vector< std::pair< std::string, std::string >> blah_items;
-//			toolkit_->get_palette_manager()->active_palette().expand_palette_groups_menu(blah_items);
-//			int selected = 1; //toolkit_->get_palette_manager()->active_palette().get_selected;
-//			gui2::teditor_select_palette_group::execute(selected, blah_items);
-		}
```

Comment:
Why would the enum be removed? There are a bunch of others that only return true as well.

Reply:
I don't see another case of just returning true. What would be the point of that?

Reply:
I mean, right below it is https://github.com/wesnoth/wesnoth/blob/master/src/editor/controller/editor_controller.cpp#L277-L292, and you can scroll down the file to find multiple other examples. My understanding is that the return value here is just to evaluate whether the function it's tied to can currently be used and having just `return true` means it can always be used.

Reply:
You're looking at a different function. We're at lines 700+ here.

Reply:
Well, removing it makes it no longer possible to select a sub-group of terrain types, so I don't think removing the enum value should be done.

Reply:
To back up a bit: My suggestion is to not have this change in this PR at all. This PR is big enough and this is unrelated noise.

When doing this change it would be better to remove the complete case since without the todo comments it's just confusing why this case is there at all. As a follow up one could check if the entire entry in the hotkey list is useless but as you noticed it is not since hotkeys are also used in menu functionality. It would be good to set the entry as hidden though so it does not show up in preferences.

Reply:
The TODO comment doesn't explain why it's there either though, and the code that's commented out doesn't either since most of what it's referencing doesn't exist anymore.

Reply:
One more try: I suggest to remove https://github.com/wesnoth/wesnoth/blob/64dcc3feb76a1252116d90c1139cdb6f37b8bfa2/src/editor/controller/editor_controller.cpp#L705-L713
And I suggest to not do that in this PR since it is not related.

Reply:
I understand what you're saying, but I don't know why it makes any difference - the code was commented out when it was originally committed 10 years ago (https://github.com/wesnoth/wesnoth/commit/adbda2dd372214e975c55b0d8b6d574360dd2654) and this is also not the only place I deleted a pointless TODO comment.

Reply:
I'm fine with pushing another commit to delete the actual `case` block later, but I don't understand why it matters whether a few lines of commented out code that's never been used is deleted here.

Reply:
Anyway, changed it back.

Reply:
1) To any reviewer it matters to focus on the actual changes relevant to this PR so any additional cleanup is distracting.
1) To anyone looking at the history it is interesting to know why code is the way it is and finding code changes in unrelated commits is not helping that.
1) To anyone reading the current code it is confusing to find code that simply does nothing. While the commented out code is not super clear on what exactly the idea is it's clear that the hotkey was supposed to open some dialog. Having a case that simply does nothing is just weird so when you remove the comments there is no reason to not remove the surrounding related code as well. That surrounding code could have as well be commented out anyway.

Reply:
Not particularly related but currently confusing code I just noticed:
https://github.com/wesnoth/wesnoth/blob/64dcc3feb76a1252116d90c1139cdb6f37b8bfa2/src/editor/controller/editor_controller.cpp#L770

A leftover from c8666a8a410ff0034a2c9bc5356ebfb80a15dd83. Especially with removals it's not so easy to find the context.

Reply:
I don't care enough about it to suggest another course of action. However I do think that it makes no sense to say that the commented out code here is associated to anything at all.